engine/src/main/java/org/destinationsol/game/SolContactListener.java

```diff
@@ -55,21 +80,44 @@ public void preSolve(Contact contact, Manifold oldManifold) {
 
     @Override
     public void postSolve(Contact contact, ContactImpulse impulse) {
-        SolObject soa = (SolObject) contact.getFixtureA().getBody().getUserData();
-        SolObject sob = (SolObject) contact.getFixtureB().getBody().getUserData();
+
+        Object dataA = contact.getFixtureA().getBody().getUserData();
+        Object dataB = contact.getFixtureB().getBody().getUserData();
+
+        Vector2 collPos = contact.getWorldManifold().getPoints()[0];
+        float absImpulse = calcAbsImpulse(impulse);
+
+        if (dataA instanceof EntityRef) {
+            entitySystemManager.sendEvent(new ImpulseEvent(collPos, absImpulse), (EntityRef) dataA);
+
+            //This is a patch to smooth over contact between an Entity and a SolObject. Once every SolObject has been
```

Comment:
This sounds like it deserves a `TODO` at the start to make it formal and allow code analytics to catch on to it :-)